test/test_resume.cpp

```diff
@@ -751,19 +751,80 @@ TORRENT_TEST(zero_file_prio_deprecated)
 {
 	test_zero_file_prio(true);
 }
+
+TORRENT_TEST(backwards_compatible_resume_info_dict)
+{
+	// make sure the "info" dictionary is picked up correctly from the
+	// resume data in backwards compatible mode
+
+	std::shared_ptr<torrent_info> ti = generate_torrent();
+	entry rd;
+	rd["file-format"] = "libtorrent resume file";
+	rd["name"] = ti->name();
+	rd["info-hash"] = ti->info_hash();
+	auto metainfo = ti->metadata();
+	rd["info"] = bdecode(metainfo.get(), metainfo.get() + ti->metadata_size());
+	std::vector<char> resume_data;
+	bencode(back_inserter(resume_data), rd);
+
+	add_torrent_params atp;
+	atp.resume_data = std::move(resume_data);
+	atp.save_path = ".";
+
+	session ses;
+	torrent_handle h = ses.add_torrent(atp);
+	auto torrent = h.torrent_file();
+	TEST_CHECK(torrent->info_hash() == ti->info_hash());
+	torrent_status s = h.status();
+}
 #endif
 
+TORRENT_TEST(resume_info_dict)
+{
+	// make sure the "info" dictionary is picked up correctly from the
+	// resume data
+
+	std::shared_ptr<torrent_info> ti = generate_torrent();
+	entry rd;
+	rd["file-format"] = "libtorrent resume file";
+	rd["name"] = ti->name();
+	rd["info-hash"] = ti->info_hash();
+	auto metainfo = ti->metadata();
+	rd["info"] = bdecode(metainfo.get(), metainfo.get() + ti->metadata_size());
+	std::vector<char> resume_data;
+	bencode(back_inserter(resume_data), rd);
+
+	error_code ec;
+	add_torrent_params atp = read_resume_data(resume_data, ec);
+	TEST_CHECK(atp.ti->info_hash() == ti->info_hash());
+}
+
 TORRENT_TEST(zero_file_prio)
 {
 	test_zero_file_prio();
 }
 
-void test_seed_mode(bool const file_prio, bool const pieces_have, bool const piece_prio
-	, bool const all_files_zero = false, bool const test_deprecated = false)
+enum class test_mode_t
 {
-	std::printf("test_seed_mode file_prio: %d pieces_have: %d piece_prio: %d\n"
-		, file_prio, pieces_have, piece_prio);
+	none = 0,
+	file_prio = 1,
+	pieces_have = 2,
+	piece_prio = 4,
+	all_files_zero = 8,
+	deprecated = 16
+};
 
+namespace libtorrent {
+namespace flags {
+
+template <>
+struct enable_flag_operators<test_mode_t> : std::true_type {};
+
+}
+}
+
+void test_seed_mode(test_mode_t const flags)
```

Comment:
tests should be treated as production code, and so, fixed the signature to not use raw bools.